src/climate_learn/data/tasks/forecasting.py

```diff
@@ -31,8 +32,9 @@ def setup(self) -> None:
         out_data = xr.concat(
             [self.dataset.data_dict[k] for k in self.out_vars], dim="level"
         )
-        self.inp_data: np.ndarray = inp_data.to_numpy().astype(np.float32)
-        self.out_data: np.ndarray = out_data.to_numpy().astype(np.float32)
+
+        self.inp_data: np.ndarray = inp_data[::self.subsample].to_numpy().astype(np.float32)
+        self.out_data: np.ndarray = out_data[::self.subsample].to_numpy().astype(np.float32)
```

Comment:
The problem with this way of subsampling is that, if our `self.pred_range` is not a factor of `self.subsample` then we can never retrieve the data as we have discarded all the data that honest lies at an interval of `self.subsample`.

Same goes for `self.window`. The calculation for calculating the next index would need to change in `self.window` as the next index would take as `self.subsample` forward instead of one timestamp. So `self.window` should be divided by `self.subsample` but this approach still doesn't solve the problem where this division results not in an integer.